lib/application.js

```diff
@@ -99,7 +99,7 @@ module.exports = class Application extends Emitter {
    */
 
   use(fn) {
-    debug('use %s', fn._name || fn.name || '-');
+    debug('use %s', fn && (fn._name || fn.name) || '-');
```

Comment:
I had an initial `if (fn === null || fn === undefined)`, but decided that it wasn't necessary... I also debated moving `debug` below the `if` (for type and generator check), but figured that having a debug output before throwing could be useful :)